antarest/study/storage/output_service.py

```diff
@@ -0,0 +1,540 @@
+                    name=task_name,
+                    type=[TaskType.UNARCHIVE, TaskType.ARCHIVE],
+                    status=[TaskStatus.RUNNING, TaskStatus.PENDING],
+                ),
+                RequestParameters(user=DEFAULT_ADMIN_USER),
+            )
+            if len(list(filter(lambda t: t.name in archive_task_names, study_tasks))):
+                raise TaskAlreadyRunning()
+
+        def archive_output_task(notifier: ITaskNotifier) -> TaskResult:
+            try:
+                study = self._study_service.get_study(study_id)
+                stopwatch = StopWatch()
+                self._storage_service.get_storage(study).archive_study_output(study, output_id)
+                stopwatch.log_elapsed(lambda x: logger.info(f"Output {output_id} of study {study_id} archived in {x}s"))
+                return TaskResult(
+                    success=True,
+                    message=f"Study output {study_id}/{output_id} successfully archived",
+                )
+            except Exception as e:
+                logger.warning(
+                    f"Could not archive the output {study_id}/{output_id}",
+                    exc_info=e,
+                )
+                raise e
+
+        task_id = self._task_service.add_task(
+            archive_output_task,
+            task_name,
+            task_type=TaskType.ARCHIVE,
+            ref_id=study.id,
+            progress=None,
+            custom_event_messages=None,
+            request_params=params,
+        )
+
+        return task_id
+
+    def aggregate_output_data(
+        self,
+        uuid: str,
+        output_id: str,
+        query_file: MCIndAreasQueryFile | MCAllAreasQueryFile | MCIndLinksQueryFile | MCAllLinksQueryFile,
+        frequency: MatrixFrequency,
+        columns_names: Sequence[str],
+        ids_to_consider: Sequence[str],
+        params: RequestParameters,
+        mc_years: Optional[Sequence[int]] = None,
+    ) -> pd.DataFrame:
+        """
+        Aggregates output data based on several filtering conditions
+
+        Args:
+            uuid: study uuid
+            output_id: simulation output ID
+            query_file: which types of data to retrieve: "values", "details", "details-st-storage", "details-res", "ids"
+            frequency: yearly, monthly, weekly, daily or hourly.
+            columns_names: regexes (if details) or columns to be selected, if empty, all columns are selected
+            ids_to_consider: list of areas or links ids to consider, if empty, all areas are selected
+            params: request parameters
+            mc_years: list of monte-carlo years, if empty, all years are selected (only for mc-ind)
+
+        Returns: the aggregated data as a DataFrame
+
+        """
+        study = self._study_service.get_study(uuid)
+        assert_permission(params.user, study, StudyPermissionType.READ)
+        study_path = self._storage_service.raw_study_service.get_study_path(study)
```

Comment:
Actually the design seems broken here and in the aggregate manager:
- the aggregate manager takes a `study_path` and then relies on the fact that outputs are stored in `study_path / output` in all cases, but it should not need to assume this. It would be better to provide him the path to the specific output we are working on, instead
- then IOutputStorage could have a method get_output_path() (which would indeed return study_path / output / <output_id> for both implementations but we must not rely on it)

This will avoid to reference explicitly the raw and variant storages in the whole class.
Then we could only have a dependency to an `OutputStorageProvider` in the constructor, which would make this service dependencies cleaner.